### PR TITLE
[202012][BRCM SAI 4.3.5.3-5] Update saibcm for pcbb feature

### PR DESCRIPTION
**Pull request by richardyu-ms**
#### Why I did it
Support Tunnel PFC/pcbb feature on Broadcom platform.

#### How to verify it
Tested build target, **successful**
```
make target/docker-syncd-brcm.gz
```
manual run those tests after installing sai binary within image 20201231.67 on 7050CX3 (TD3) T0 DUT, **all passed**
```
     fib/test_fib.py
     vxlan/test_vxlan_decap.py
     fdb/test_fdb.py
     decap/test_decap.py
     pfcwd/test_pfcwd_all_port_storm.py
     acl/null_route/test_null_route_helper.py
     acl/test_acl.py
     vlan/test_vlan.py
     platform_tests/test_reboot.py
```

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [x] 202012
- [ ] 202106
- [ ] 202111

#### Description for the changelog

[BRCM SAI 4.3.5.3-4] Support Tunnel PFC/pcbb in 202012

### CODE REVIEW

**Comment by lguohan** [comment]
can you check other sai.mk pr, please describe the test you have done with this sai pacakge update.

File: platform/broadcom/sai.mk [hunk 1]
```diff
@@ -1,9 +1,18 @@
...
+
+BRCM_SAI = libsaibcm_$(LIBSAIBCM_VERSION)_amd64.deb
+$(BRCM_SAI)_URL = "$(LIBSAIBCM_URL_PREFIX)/$(BRCM_SAI)"
+BRCM_SAI_DEV = libsaibcm-dev_$(LIBSAIBCM_VERSION)_amd64.deb
 $(eval $(call add_derived_package,$(BRCM_SAI),$(BRCM_SAI_DEV)))
-$(BRCM_SAI_DEV)_URL = "https://sonicstorage.blob.core.windows.net/packages/bcmsai/4.3/202012/libsaibcm-dev_4.3.5.3-3_amd64.deb?sv=2020-04-08&st=2022-03-16T21%3A33%3A39Z&se=2037-03-17T21%3A33%3A00Z&sr=b&sp=r&sig=n5nodstTKbsJLZDjhQYmpFB3rafcYPybsMaici4%2FwfE%3D"
+$(BRCM_SAI_DEV)_URL = "$(LIBSAIBCM_URL_PREFIX)/$(BRCM_SAI_DEV)"
+
+# SAI module for DNX Asic family
+BRCM_DNX_SAI = libsaibcm_dnx_$(LIBSAIBCM_VERSION)_amd64.deb
```
Comment (lguohan) [inline platform/broadcom/sai.mk:12]: i am not sure we want to make such change in 202012 branch.

File: platform/broadcom/sai.mk [hunk 2]
```diff
@@ -1,9 +1,18 @@
-BRCM_SAI = libsaibcm_4.3.5.3-3_amd64.deb
-$(BRCM_SAI)_URL = "https://sonicstorage.blob.core.windows.net/packages/bcmsai/4.3/202012/libsaibcm_4.3.5.3-3_amd64.deb?sv=2020-04-08&st=2022-03-16T21%3A31%3A49Z&se=2037-03-17T21%3A31%3A00Z&sr=b&sp=r&sig=6meWjaM%2FMyQRJIigjqs4xc7ZCGotP0S94np26kn9zls%3D"
-BRCM_SAI_DEV = libsaibcm-dev_4.3.5.3-3_amd64.deb
+LIBSAIBCM_VERSION = 4.3.5.3-4
```
Comment (gechiang) [inline platform/broadcom/sai.mk:1]: For 4.3 we can not do it like other newer branches such as Master.  There is no automation support for 20201231 branch so please continue to use URL for now as before for the new version.
Please take a look all previous SAI 4.3.5 PRs as model on what you need to do to make change in this file.
also there is a need to perform the following to pick up another file that is needed for this PR:

'''

	make target/debs/buster/libsaibcm_4.3.5.3-4_amd64.deb   <== replace with the correct version so for you it would be the 4.3.5.3-4
	make freeze 
	git status
	There should be 2 files you need to include for your PR: "sai.mk" and "files/build/versions/default/versions-web"
'''

File: platform/broadcom/sai.mk (hunk 1 above)
Comment (richardyu-ms) [reply platform/broadcom/sai.mk:12]: removed DNX part

File: platform/broadcom/sai.mk (hunk 2 above)
Comment (richardyu-ms) [reply platform/broadcom/sai.mk:1]: tested with change from 4.3.5.3-2 to 4.3.5.3-3(on commit:decb3bfc4 just ahead of the change for updating to 4.3.5.3-3). after ``make freeze``, there are 7 files changed. following that change just take the change in version-web.

Also, compare w/o the change in version-web, for building syncd, both the scenarios can build successful

File: files/build/versions/default/versions-web [hunk 3]
```diff
@@ -51,6 +51,8 @@ https://sonicstorage.blob.core.windows.net/debian/pool/main/n/net-snmp/net-snmp_
 https://sonicstorage.blob.core.windows.net/debian/pool/main/n/net-snmp/net-snmp_5.7.3+dfsg-5.dsc==2443e4dffbdb020e7ab4f947a7904912
 https://sonicstorage.blob.core.windows.net/debian/pool/main/n/net-snmp/net-snmp_5.7.3+dfsg.orig.tar.xz==6391ae27eb1ae34ff5530712bb1c4209
 https://sonicstorage.blob.core.windows.net/packages/20190307/dsserve?sv=2015-04-05&sr=b&sig=lk7BH3DtW%2F5ehc0Rkqfga%2BUCABI0UzQmDamBsZH9K6w%3D&se=2038-05-06T22%3A34%3A45Z&sp=r==f9d4b815ebb9be9f755dedca8a51170d
+https://sonicstorage.blob.core.windows.net/packages/bcmsai/4.3/202012/libsaibcm-dev_4.3.5.3-3_amd64.deb?sv=2020-04-08&st=2022-03-16T21%3A33%3A39Z&se=2037-03-17T21%3A33%3A00Z&sr=b&sp=r&sig=n5nodstTKbsJLZDjhQYmpFB3rafcYPybsMaici4%2FwfE%3D==cf1693725815b6dc2f4a5c7c48f7088d
```
Comment (gechiang) [inline files/build/versions/default/versions-web:54]: this is from the old version.  this needs to be manually deleted.

File: files/build/versions/default/versions-web [hunk 4]
```diff
@@ -51,6 +51,8 @@ https://sonicstorage.blob.core.windows.net/debian/pool/main/n/net-snmp/net-snmp_
...
+https://sonicstorage.blob.core.windows.net/packages/bcmsai/4.3/202012/libsaibcm_4.3.5.3-3_amd64.deb?sv=2020-04-08&st=2022-03-16T21%3A31%3A49Z&se=2037-03-17T21%3A31%3A00Z&sr=b&sp=r&sig=6meWjaM%2FMyQRJIigjqs4xc7ZCGotP0S94np26kn9zls%3D==3de43b33c96c355f4b69a911f1e7f926
```
Comment (gechiang) [inline files/build/versions/default/versions-web:55]: so is this old one needs to be removed.

File: files/build/versions/default/versions-web [hunk 5]
```diff
@@ -71,6 +73,8 @@ https://sonicstorage.blob.core.windows.net/packages/redis/redis-server_6.0.6-1~b
 https://sonicstorage.blob.core.windows.net/packages/redis/redis-tools_6.0.6-1_bpo10+1_arm64.deb?sv=2015-04-05&sr=b&sig=GbkJV2wWln3hoz27zKi5erdk3NDKrAFrQriA97bcRCY%3D&se=2030-10-24T04%3A22%3A21Z&sp=r==282b4766cc9ac7d8bb70622bd69d9f5c
 https://sonicstorage.blob.core.windows.net/packages/redis/redis-tools_6.0.6-1_bpo10+1_armhf.deb?sv=2015-04-05&sr=b&sig=67vHAMxsl%2BS3X1KsqhdYhakJkGdg5FKSPgU8kUiw4as%3D&se=2030-10-24T04%3A22%3A40Z&sp=r==62f287117afab6caaec564232ebbb5de
 https://sonicstorage.blob.core.windows.net/packages/redis/redis-tools_6.0.6-1~bpo10+1_amd64.deb?sv=2015-04-05&sr=b&sig=73zbmjkf3pi%2Bn0R8Hy7CWT2EUvOAyzM5aLYJWCLySGM%3D&se=2030-09-06T19%3A44%3A59Z&sp=r==2d58c3c3358290c04d5e0ba70f297f18
+https://sonicstorage.blob.core.windows.net/public/sai/bcmsai/REL_4.3.5.2/4.3.5.3-4/libsaibcm-dev_4.3.5.3-4_amd64.deb==0cc17be6aad8d0c470567d425b432d5f
```
Comment (gechiang) [inline files/build/versions/default/versions-web:76]: this one is new but URL looks not correct.  will ping you offline.

File: files/build/versions/default/versions-web [hunk 6]
```diff
@@ -71,6 +73,8 @@ https://sonicstorage.blob.core.windows.net/packages/redis/redis-server_6.0.6-1~b
...
+https://sonicstorage.blob.core.windows.net/public/sai/bcmsai/REL_4.3.5.2/4.3.5.3-4/libsaibcm_4.3.5.3-4_amd64.deb==5e2cbd5c6b4843e50642d38569881243
```
Comment (gechiang) [inline files/build/versions/default/versions-web:77]: same as above... URL may not be correct.

File: platform/broadcom/sai.mk [hunk 7]
```diff
@@ -1,8 +1,8 @@
-BRCM_SAI = libsaibcm_4.3.5.3-3_amd64.deb
-$(BRCM_SAI)_URL = "https://sonicstorage.blob.core.windows.net/packages/bcmsai/4.3/202012/libsaibcm_4.3.5.3-3_amd64.deb?sv=2020-04-08&st=2022-03-16T21%3A31%3A49Z&se=2037-03-17T21%3A31%3A00Z&sr=b&sp=r&sig=6meWjaM%2FMyQRJIigjqs4xc7ZCGotP0S94np26kn9zls%3D"
-BRCM_SAI_DEV = libsaibcm-dev_4.3.5.3-3_amd64.deb
+BRCM_SAI = libsaibcm_4.3.5.3-4_amd64.deb
+$(BRCM_SAI)_URL = "https://sonicstorage.blob.core.windows.net/public/sai/bcmsai/REL_4.3.5.2/4.3.5.3-4/libsaibcm_4.3.5.3-4_amd64.deb"
```
Comment (gechiang) [inline platform/broadcom/sai.mk:2]: the URL seems missing proper read permission date. will ping you offline

**Review by gechiang — COMMENTED** [review]
Will ping you offline to make the correction

File: files/build/versions/default/versions-web [hunk 8]
```diff
@@ -71,6 +74,8 @@ https://sonicstorage.blob.core.windows.net/packages/redis/redis-server_6.0.6-1~b
...
```
[the remark this answers is not in the document]
Comment (gechiang) [reply files/build/versions/default/versions-web:78]: I think what happened with this is that when you started with the wrong url in your previous change, it somehow dirtied it even after you performed make clean.  My suggestion would be for you to delete your local sonic-buildimage and re-clone it and then add the sai.mk change and then perform the build and freeze and see if that makes a difference.

**Review by gechiang — CHANGES_REQUESTED** [review]

**Review by richardyu-ms — COMMENTED** [review]
re-build after make reset and git clean -xdf.
update the package url as well

File: platform/broadcom/sai.mk [hunk 9]
```diff
@@ -1,8 +1,8 @@
-BRCM_SAI = libsaibcm_4.3.5.3-3_amd64.deb
-$(BRCM_SAI)_URL = "https://sonicstorage.blob.core.windows.net/packages/bcmsai/4.3/202012/libsaibcm_4.3.5.3-3_amd64.deb?sv=2020-04-08&st=2022-03-16T21%3A31%3A49Z&se=2037-03-17T21%3A31%3A00Z&sr=b&sp=r&sig=6meWjaM%2FMyQRJIigjqs4xc7ZCGotP0S94np26kn9zls%3D"
-BRCM_SAI_DEV = libsaibcm-dev_4.3.5.3-3_amd64.deb
+BRCM_SAI = libsaibcm_4.3.5.3-5_amd64.deb
+$(BRCM_SAI)_URL = "https://sonicstorage.blob.core.windows.net/packages/bcmsai/4.3/202012/libsaibcm-dev_4.3.5.3-5_amd64.deb?sv=2020-08-04&st=2022-06-06T04%3A45%3A06Z&se=2037-06-07T04%3A45%3A00Z&sr=b&sp=r&sig=SF6YOtYzSIcwSR4BRInnimUjbusNp4mPHFunFjm%2Fors%3D"
```
Comment (lguohan) [inline platform/broadcom/sai.mk:2]: this is dev package?

File: files/build/versions/default/versions-web [hunk 10]
```diff
@@ -51,6 +51,8 @@ https://sonicstorage.blob.core.windows.net/debian/pool/main/n/net-snmp/net-snmp_
 https://sonicstorage.blob.core.windows.net/debian/pool/main/n/net-snmp/net-snmp_5.7.3+dfsg-5.dsc==2443e4dffbdb020e7ab4f947a7904912
 https://sonicstorage.blob.core.windows.net/debian/pool/main/n/net-snmp/net-snmp_5.7.3+dfsg.orig.tar.xz==6391ae27eb1ae34ff5530712bb1c4209
 https://sonicstorage.blob.core.windows.net/packages/20190307/dsserve?sv=2015-04-05&sr=b&sig=lk7BH3DtW%2F5ehc0Rkqfga%2BUCABI0UzQmDamBsZH9K6w%3D&se=2038-05-06T22%3A34%3A45Z&sp=r==f9d4b815ebb9be9f755dedca8a51170d
+https://sonicstorage.blob.core.windows.net/packages/bcmsai/4.3/202012/libsaibcm-dev_4.3.5.3-5_amd64.deb?sv=2020-08-04&st=2022-06-06T04%3A45%3A06Z&se=2037-06-07T04%3A45%3A00Z&sr=b&sp=r&sig=SF6YOtYzSIcwSR4BRInnimUjbusNp4mPHFunFjm%2Fors%3D==e9861d9d73d17207e120b2c089ae590a
+https://sonicstorage.blob.core.windows.net/packages/bcmsai/4.3/202012/libsaibcm_4.3.5.3-5_amd64.deb?sv=2020-08-04&st=2022-06-06T04%3A44%3A45Z&se=2037-06-07T04%3A44%3A00Z&sr=b&sp=r&sig=uG51Q6rJrZYNVVuoShw8F4zas1n6zi%2FJ7udpTSZ87M8%3D==1594e9cfb991e0ed2d610d84a141a731
```
Comment (lguohan) [inline files/build/versions/default/versions-web:55]: why is this needed?

File: platform/broadcom/sai.mk (hunk 9 above)
Comment (richardyu-ms) [reply platform/broadcom/sai.mk]: fixed

File: files/build/versions/default/versions-web (hunk 10 above)
Comment (xumia) [reply files/build/versions/default/versions-web:55]: It is to lock the hash of the download file, prevent the file with the same name changed in the remote web server.
It is a good practice to add/change the lock file, but it is not required. The automation job will lock the hash if not changed.

**Review by gechiang — APPROVED** [review]
New changes look good to me.

**Comment by gechiang** [comment]
Added @vmittal-msft as reviewer to ensure that this change is not missing some important changes required from BRCM for PFCWD support.
